ssl/ssl_lib.c

```diff
@@ -4118,6 +4169,42 @@ SSL_CTX *SSL_CTX_new_ex(OSSL_LIB_CTX *libctx, const char *propq,
         goto err;
     }
 
+#ifndef OPENSSL_NO_SSLKEYLOG
+    /* Make sure we have a global lock allocated */
+    if (!RUN_ONCE(&ssl_keylog_once, ssl_keylog_init)) {
+        /* log an error, but don't fail here */
+        ERR_raise(ERR_LIB_SSL, ERR_R_SSL_LIB);
+    } else {
+        /* If someone set SSLKEYLOGFILE ... */
+        if (keylogfile != NULL) {
+            /* Grab out global lock */
+            if (!CRYPTO_THREAD_write_lock(keylog_lock)) {
+                ERR_raise(ERR_LIB_SSL, ERR_R_SSL_LIB);
+            } else {
+                /*
+                 * If the bio for the requested keylog file hasn't been
+                 * created yet, go ahead and create it, truncating it to zero
```

Comment:
This is unusual behaviour. Generally you want to append to this file so that multiple applications, or multiple invocations of the same application, can share the same logfile.

Reply:
I was concerned about this.  Most notably I was concerned that, in the former case (multiple applications) sharing a keylog file might leak secrets to users that shouldn't have them, ie:
SSLKEYLOGFILE=/tmp/keys.log <application>
when  user A and B might expose each users's secrets to the other.

Or is this a situation where this is inherently insecure, and you get what you get?

Reply:
addressed to make this an append operation, based on the assumption that keylogging is inherently insecure